libsolidity/interface/StandardCompiler.cpp

```diff
@@ -605,7 +605,27 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 		if (details.isMember("yulDetails"))
 		{
 			if (!settings.runYulOptimiser)
+			{
+				auto const& yulDetails = details["yulDetails"];
+				if (yulDetails.isObject() && yulDetails.isMember("optimizerSteps"))
+				{
+					auto error = checkOptimizerDetailSteps(details["yulDetails"], "optimizerSteps", settings.yulOptimiserSteps, settings.yulOptimiserCleanupSteps);
+					auto isWhiteSpaceOrNewlineOnly = [](std::string const& _sequence) {
+						return !std::any_of(_sequence.begin(), _sequence.end(), [](char const step){
+							return step != ' ' && step != '\n';
+						});
+					};
+					if (
+						!error &&
+						isWhiteSpaceOrNewlineOnly(settings.yulOptimiserSteps) &&
+						isWhiteSpaceOrNewlineOnly(settings.yulOptimiserCleanupSteps)
+					)
+						return { std::move(settings) };
+					else
+						return formatFatalError(Error::Type::JSONError, "\"If Yul optimizer is disabled, only an empty optimizerSteps sequence (\":\") is accepted.\"");
+				}
 				return formatFatalError(Error::Type::JSONError, "\"Providing yulDetails requires Yul optimizer to be enabled.");
```

Comment:
What's up with the extra quotes here?
```suggestion
						return formatFatalError(Error::Type::JSONError, "If Yul optimizer is disabled, only an empty optimizerSteps sequence (\":\") is accepted.");
				}
				return formatFatalError(Error::Type::JSONError, "Providing yulDetails requires Yul optimizer to be enabled.");
```